runtime/doc/diagnostic.txt

```diff
@@ -570,42 +560,43 @@ get_prev({opts})                                   *vim.diagnostic.get_prev()*
+    Move to a diagnostic.
 
     Parameters: ~
-      • {opts}  (table|nil) Configuration table with the following keys:
+      • {opts}  vim.diagnostic.GotoOptions A table with the following optional keys:
+                • diagnostic: (table) The diagnostic item to move to
+                  |diagnostic-structure|. Mutually exclusive with "count",
+                  "namespace", and "severity".
+                • count: (number) The number of diagnostics to move by. A
+                  positive integer moves forward by {count} diagnostics, while
+                  a negative integer moves backward by {count} diagnostics.
+                  Mutually exclusive with "diagnostic".
                 • namespace: (number) Only consider diagnostics from the given
-                  namespace.
-                • cursor_position: (cursor position) Cursor position as a
-                  (row, col) tuple. See |nvim_win_get_cursor()|. Defaults to
-                  the current cursor position.
-                • wrap: (boolean, default true) Whether to loop around file or
-                  not. Similar to 'wrapscan'.
-                • severity: See |diagnostic-severity|.
-                • float: (boolean or table, default true) If "true", call
-                  |vim.diagnostic.open_float()| after moving. If a table, pass
-                  the table as the {opts} parameter to
+                  namespace. Only used when "count" is non-nil.
+                • cursor_position: (table) Cursor position as a (row, col)
+                  tuple. Used to find the nearest diagnostic when "count" is
+                  used. Only used when "count" is non-nil.
+                • wrap: (boolean) Whether to wrap around the file or not.
+                  Similar to 'wrapscan'. Default true.
+                • severity: Only consider diagnostics that match the given
+                  |diagnostic-severity|. Only used when "count" is non-nil.
+                • float: (boolean or table) If "true", call
+                  |vim.diagnostic.open_float()| after moving. If a table, also
+                  pass the table as the {opts} parameter to
                   |vim.diagnostic.open_float()|. Unless overridden, the float
                   will show diagnostics at the new cursor position (as if
-                  "cursor" were passed to the "scope" option).
-                • win_id: (number, default 0) Window ID
+                  "cursor" were used as the "scope" option).
+                • winid: (number) Window ID. Default 0 (current window).
+
+    Return: ~
+        Diagnostic ?|nil @The diagnostic moved to, if any.
 
 goto_prev({opts})                                 *vim.diagnostic.goto_prev()*
     Move to the previous diagnostic in the current buffer.
 
     Parameters: ~
-      • {opts}  (table|nil) See |vim.diagnostic.goto_next()|
+      • {opts}  (table|nil) See |vim.diagnostic.goto_next()| deprecated
```

Comment:
This section `*vim.diagnostic.goto_prev()` should also be removed? (probably catched by CI)